bigquery_etl/query_scheduling/generate_airflow_dags.py

```diff
@@ -95,16 +95,41 @@ def get_dags(project_id, dags_config):
                 else:
                     if CHECKS_FILE in files:
                         checks_file = os.path.join(root, CHECKS_FILE)
-                        checks_task = copy.deepcopy(
-                            Task.of_dq_check(checks_file, dag_collection=dag_collection)
-                        )
-                        tasks.append(checks_task)
-                        task_ref = TaskRef(
-                            dag_name=task.dag_name,
-                            task_id=task.task_name,
-                        )
-                        checks_task.upstream_dependencies.append(task_ref)
+                        with open(checks_file, "r") as file:
+                            file_contents = file.read()
+                        # check if file contains fail and warn and create checks task accordingly
+                        if "#fail" in file_contents:
+                            is_check_task_fail = True
+                            checks_fail_task = copy.deepcopy(
+                                Task.of_dq_check(
+                                    checks_file,
+                                    is_check_task_fail,
+                                    dag_collection=dag_collection,
+                                )
+                            )
+                            tasks.append(checks_fail_task)
+                            task_ref = TaskRef(
+                                dag_name=task.dag_name,
+                                task_id=task.task_name,
+                            )
+                            checks_fail_task.upstream_dependencies.append(task_ref)
+                        if "#warn" in file_contents:
+                            is_check_task_fail = False
+                            checks_warn_task = copy.deepcopy(
+                                Task.of_dq_check(
+                                    checks_file,
+                                    is_check_task_fail,
+                                    dag_collection=dag_collection,
+                                )
+                            )
+                            tasks.append(checks_warn_task)
+                            task_ref = TaskRef(
+                                dag_name=task.dag_name,
+                                task_id=task.task_name,
+                            )
+                            checks_warn_task.upstream_dependencies.append(task_ref)
```

Comment:
We could probably rewrite this a bit to something like:

```suggestion
                        is_check_task_fail = True
                        if "#warn" in file_contents:
                             is_check_task_fail = False
                             
                         checks_task = copy.deepcopy(
                                Task.of_dq_check(
                                    checks_file,
                                    is_check_task_fail,
                                    dag_collection=dag_collection,
                                )
                            )
                            tasks.append(checks_task)
                            task_ref = TaskRef(
                                dag_name=task.dag_name,
                                task_id=task.task_name,
                            )
                            checks_task.upstream_dependencies.append(task_ref)
                           
```

This will always create a task for `fail` even if there are none specified, but I think that's alright for now